src/components/Store.tsx

```diff
@@ -191,6 +198,15 @@ function reducer(state: ChatState, action: A): ChatState {
                         ? state.roomIds.filter((id) => id !== action.payload)
                         : state.roomIds,
             }
+
+        case ActionType.SetRoomMembers:
+            const roomMembers = state.roomMembers
+            roomMembers[action.payload.roomId] = [...action.payload.members]
+
+            return {
+                ...state,
+                roomMembers,
```

Comment:
`roomMembers` reference does not change. To React it's the same object before and after. I mean,

```ts
const { roomMembers } = useStore() 
```

…is not gonna cause a re-render after `SetRoomMembers` action gets executed.

To do it properly (so that views can detect a change) you'd have to recreate it with your mods

```ts
return {
    ...state,
    roomMembers: {
        ...state.roomMembers,
        [action.payload.roomId]: [...action.payload.members],
    }
}
```